pkg/service/collector.go

```diff
@@ -89,3 +89,35 @@ func GetNameForCollectorService(jaeger *v1.Jaeger) string {
 func GetNameForHeadlessCollectorService(jaeger *v1.Jaeger) string {
 	return util.DNSName(util.Truncate("%s-collector-headless", 63, jaeger.Name))
 }
+
+// GetPortNameForGRPC returns the port name for 'grpc'. It may either be http-grpc or https-grpc, based on whether
+// TLS is enabled for the agent-collector gRPC communication
+func GetPortNameForGRPC(jaeger *v1.Jaeger) string {
+	if viper.GetString("platform") == v1.FlagPlatformOpenShift {
+		// we always have TLS certs when running on OpenShift, so, TLS is always enabled
+		return "https-grpc"
+	}
+
+	// if we don't have a jaeger provided, it's certainly not TLS...
+	if nil == jaeger {
```

Comment:
Should this ever happen?

Reply:
Not in the common path, not with the code we have today.